src/vectorchord/advanced-features/pre-and-post-filtering.md

```diff
@@ -0,0 +1,57 @@
+# Prefiltering And Postfiltering <badge type="tip" text="since v0.4.0" />
+
+VectorChord provides flexible filtering mechanisms to improve vector search performance and accuracy through **prefiltering** and **postfiltering** strategies. These filtering approaches determine when query conditions (like `WHERE` clauses) are applied during the vector similarity search process.
+
+Unlike **postfiltering**, which is the default behavior, the **prefiltering** strategy applies an additional filter before the vector search. This prunes the search space, thereby increasing the speed of the search. However, the additional filter also introduces an extra cost that should be considered.
+
+Only a `filter on Index Scan` can benefit from prefiltering:
+
+```sql
+EXPLAIN (COSTS FALSE, TIMING FALSE) 
+SELECT val FROM t WHERE tag > 1 ORDER BY val <-> '[0, 0, 0]';
+               QUERY PLAN                
+-----------------------------------------
+ Index Scan using t_val_idx on t
+   Order By: (val <-> '[0,0,0]'::vector)
+   Filter: (tag > 1)
+```
+
+## Configuration
+
+You can control the filtering strategy using the `vchordrq.prefilter` setting:
+
+```sql
+-- Enable prefiltering (default: off)
+SET vchordrq.prefilter = on;
+```
+
+:::warning
+The [planner/optimizer](https://www.postgresql.org/docs/current/planner-optimizer.html) greatly affects whether the prefiltering takes effect. Complex queries, such as join queries, are more likely to exhibit unstable behavior. Please evaluate carefully based on experiments.
```

Comment:
It’s already been made clear that this is a filter on the `IndexScan`. How could there still be a planner/optimizer issue?